src/neuroglancer/datasource/render/frontend.ts

```diff
@@ -0,0 +1,345 @@
+}
+
+function parseStackName(stackIdObj: any): string {
+  verifyObject(stackIdObj);
+  return verifyObjectProperty(stackIdObj, 'stack', verifyString);
+}
+
+function parseStackOwner(stackIdObj: any): string {
+  verifyObject(stackIdObj);
+  return verifyObjectProperty(stackIdObj, 'owner', verifyString);
+}
+
+function parseStackInfo(obj: any): StackInfo {
+  verifyObject(obj);
+  let lowerVoxelBound: Vec3 = verifyObjectProperty(obj, 'stats', parseLowerVoxelBounds);
+  let upperVoxelBound: Vec3 = verifyObjectProperty(obj, 'stats', parseUpperVoxelBounds);
+
+  let voxelResolution: Vec3 = verifyObjectProperty(obj, 'currentVersion', parseStackVersionInfo);
+
+  let mipMapLevels: number =
+      verifyObjectProperty(obj, 'currentMipmapPathBuilder', parseMipMapLevels);
+
+  let project: string = verifyObjectProperty(obj, 'stackId', parseStackProject);
+
+  return {lowerVoxelBound, upperVoxelBound, voxelResolution, mipMapLevels, project};
+}
+
+function parseUpperVoxelBounds(stackStatsObj: any): Vec3 {
+  verifyObject(stackStatsObj);
+  let stackBounds = verifyObjectProperty(stackStatsObj, 'stackBounds', verifyObject);
+
+  let upperVoxelBound: Vec3 = vec3.create();
+
+  upperVoxelBound[0] = verifyObjectProperty(stackBounds, 'maxX', verifyInt);
+  upperVoxelBound[1] = verifyObjectProperty(stackBounds, 'maxY', verifyInt);
+  upperVoxelBound[2] = verifyObjectProperty(stackBounds, 'maxZ', verifyInt);
+
+  for (let i = 0; i < 3; i++) {
+    upperVoxelBound[i] += 1;
+  }
+
+  return upperVoxelBound;
+}
+
+function parseLowerVoxelBounds(stackStatsObj: any): Vec3 {
+  verifyObject(stackStatsObj);
+  let stackBounds = verifyObjectProperty(stackStatsObj, 'stackBounds', verifyObject);
+
+  let lowerVoxelBound: Vec3 = vec3.create();
+
+  lowerVoxelBound[0] = verifyObjectProperty(stackBounds, 'minX', verifyInt);
+  lowerVoxelBound[1] = verifyObjectProperty(stackBounds, 'minY', verifyInt);
+  lowerVoxelBound[2] = verifyObjectProperty(stackBounds, 'minZ', verifyInt);
+
+  return lowerVoxelBound;
+}
+
+function parseStackVersionInfo(stackVersionObj: any): Vec3 {
+  verifyObject(stackVersionObj);
+  let voxelResolution: Vec3 = vec3.create();
+  try {
+    voxelResolution[0] = verifyObjectProperty(stackVersionObj, 'stackResolutionX', verifyFloat);
+    voxelResolution[1] = verifyObjectProperty(stackVersionObj, 'stackResolutionY', verifyFloat);
+    voxelResolution[2] = verifyObjectProperty(stackVersionObj, 'stackResolutionZ', verifyFloat);
+  } catch (Error) {
+    // default is 1, 1, 1
+    voxelResolution[0] = 1;
+    voxelResolution[1] = 1;
+    voxelResolution[2] = 1;
+  }
+
+  return voxelResolution;
+}
+
+function parseMipMapLevels(currentMipMapPathBuilderObj: any): number {
+  let levels = 0;
+  try {
+    levels = verifyObjectProperty(currentMipMapPathBuilderObj, 'numberOfLevels', verifyInt);
+  } catch (Error) {
+    // TODO: Something better than console.log for passing messages?
+    console.log('No Mip Map Levels specified. Using default of 0.');
```

Comment:
An info button next to the Help button (say an "!" in addition to the "?") that displays a popup of log messages might do the trick. Specifically, it might be nice to allow the user to try to reload a given tile/volume/chunk in the event of a load failure. We sometimes encounter those in render and ndstore, e.g. if there's a cache miss that times out. 

As far as the Mip Map Levels error message... I think I've figured out how to properly handle Mip Map levels, so this should go away. I'll write that up in a separate post.